### ISSUE TITLE
Rounding error resulting in slightly incorrect Locked Gold balance

### ISSUE THREAD

**Issue by jamescowling** [celo-org/celo-monorepo#3843]
### Expected Behavior

I locked 10,000 Gold and am running a validator on it. It should stay as 10,000 locked gold.

### Current Behavior

```
$ celocli lockedgold:show 0x0EBdCCD9091EFB1243417bDf3aDdd63132962586
Running Checks:
   ✔  0x0EBdCCD9091EFB1243417bDf3aDdd63132962586 is a registered Account
lockedGold:
  total: 9999999999999999999994 (~10000 10^18)
  nonvoting: 0
  requirement: 10000000000000000000000 (~10000 10^18)
pendingWithdrawals:
```

**Comment by jamescowling** [comment]
I've changed my votes a few times on this address so it's possible that's triggering a rounding error somewhere.

fyi this address is a ReleaseGold contract.

**Comment by timmoreton** [comment]
Balances is as expected, but displayed incorrectly:
```
> eth.getBalance('0x0EBdCCD9091EFB1243417bDf3aDdd63132962586')
1000000000000000000
```

**Comment by jamescowling** [comment]
> Balances is as expected, but displayed incorrectly:
> 
> ```
> > eth.getBalance('0x0EBdCCD9091EFB1243417bDf3aDdd63132962586')
> 1000000000000000000
> ```

This is the free gold balance right? The issue here is with locked gold.

**Comment by jamescowling** [comment]
update from @timmoreton:
>  yes there is a rounding issue internal to how activated lockedgold is managed that may make things a few Wei off

**Comment by zviadm** [comment]
just an update here. this rounding error can cause pretty serious issues for validator groups and validators. this is because when locked balance drops below requirement, epoch rewards aren't paid out for a validator. if it drops for a group, that is even worse, because voter rewards will not be paid out at all!

so this is a pretty serious issue and can lead to loss of more significant amount of funds as it happened in practice last epoch.

**Comment by zviadm** [comment]
also this is labelled as 'cli' issue, but this isn't a client issue, this is definitely a protocol issue

**Comment by pranaymohan** [comment]
@JamesCowling @zviadm thanks for reporting -- added some details here as well FYI: https://forum.celo.org/t/epoch-rewards-issue/498

**Comment by nambrot** [comment]
Closed in #4461
